go-controller/pkg/node/secondary_node_network_controller_test.go

```diff
@@ -24,7 +33,62 @@ var _ = Describe("SecondaryNodeNetworkController", func() {
 	var (
 		nad = ovntest.GenerateNAD("bluenet", "rednad", "greenamespace",
 			types.Layer3Topology, "100.128.0.0/16", types.NetworkRolePrimary)
+		netName             = "bluenet"
+		netID               = 3
+		nodeName     string = "worker1"
+		mgtPortMAC   string = "00:00:00:55:66:77"
+		fexec        *ovntest.FakeExec
+		testNS       ns.NetNS
+		vrf          *vrfmanager.Controller
+		v4NodeSubnet = "10.128.0.0/24"
+		v6NodeSubnet = "ae70::66/112"
+		mgtPort      = fmt.Sprintf("%s%d", types.K8sMgmtIntfNamePrefix, netID)
+		stopCh       chan struct{}
+		wg           *sync.WaitGroup
+		kubeMock     kubemocks.Interface
 	)
+	BeforeEach(func() {
+		// Restore global default values before each testcase
+		Expect(config.PrepareTestConfig()).To(Succeed())
+		// Set up a fake vsctl command mock interface
+		kubeMock = kubemocks.Interface{}
+		fexec = ovntest.NewFakeExec()
+		err := util.SetExec(fexec)
+		Expect(err).NotTo(HaveOccurred())
+		// Set up a fake k8sMgmt interface
+		testNS, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+		err = testNS.Do(func(ns.NetNS) error {
+			defer GinkgoRecover()
+			ovntest.AddLink(mgtPort)
+			return nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+		wg = &sync.WaitGroup{}
+		stopCh = make(chan struct{})
+		vrf = vrfmanager.NewController()
+		wg2 := &sync.WaitGroup{}
+		defer func() {
+			wg2.Wait()
+		}()
+		wg2.Add(1)
+		go testNS.Do(func(netNS ns.NetNS) error {
+			defer wg2.Done()
+			defer GinkgoRecover()
+			err = vrf.Run(stopCh, wg)
```

Comment:
Id rather if you called the func undertest here instead of Run that creates a go routine that escapes this netns sandbox.

Reply:
see my other reply and comment.